server/src/main/java/io/druid/client/selector/ServerSelector.java

```diff
@@ -62,59 +66,88 @@ public void addServerAndUpdateSegment(
   {
     synchronized (this) {
       this.segment.set(segment);
-      int priority = server.getServer().getPriority();
-      Set<QueryableDruidServer> priorityServers = servers.computeIfAbsent(priority, p -> new HashSet<>());
+      Set<QueryableDruidServer> priorityServers = new HashSet<>();
+      if (server.getServer().getType() == ServerType.REALTIME) {
+        priorityServers = realtimeServers.computeIfAbsent(
+            server.getServer().getPriority(), p -> new HashSet<>());
```

Comment:
It's not according to Druid's style guide. If the first method argument is on the next line, each should be on it's own line.

So it should be
```java
realtimeServers.computeIfAbsent(
    server.getServer().getPriority(),
    p -> new HashSet<>()
);
```
or 
```java
realtimeServers
    .computeIfAbsent(server.getServer().getPriority(), p -> new HashSet<>());
```